### PR TITLE
 support multi-batch test in profile tool

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Support `batch_size > 1` for `tools/profile.py`, requested in issue #861 . 

## Modification

- update `create_input` for batch images in mmcls, mmdet, mmpose, mmseg, mmocr, mmrotate.
- update `tools/profile.py` to support testing with `batch_size>1`

### tested models with tensorrt backend
mmcls: resnet
mmdet: yolov3, maskrcnn
mmseg: pspnet
mmocr: dbnet, crnn
mmedit: srcnn
mmrotate: rotated_retinanet

## BC-breaking (Optional)

None

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by AllentDan — COMMENTED** [review]
Just for reminding, `model` and `shape` seem to be a must for `profile.py` why not remove `--`.
```
$ python tools/profile.py -h                                                                                               
usage: profile.py [-h] [--model MODEL [MODEL ...]] [--device DEVICE]
                  [--shape SHAPE] [--warmup WARMUP] [--num-iter NUM_ITER]
                  [--cfg-options CFG_OPTIONS [CFG_OPTIONS ...]]
                  deploy_cfg model_cfg image_dir

MMDeploy Model Latency Test Tool.

positional arguments:
  deploy_cfg            Deploy config path
  model_cfg             Model config path
  image_dir             Input directory to image files

optional arguments:
  -h, --help            show this help message and exit
  --model MODEL [MODEL ...]
                        Input model files.
  --device DEVICE       device type for inference
  --shape SHAPE         Input shape to test in `HxW` format, e.g., `800x1344`
  --warmup WARMUP       warmup iterations before counting inference latency.
  --num-iter NUM_ITER   Number of iterations to run the inference.
  --cfg-options CFG_OPTIONS [CFG_OPTIONS ...]
                        override some settings in the used config, the key-
                        value pair in xxx=yyy format will be merged into
                        config file. If the value to be overwritten is a list,
                        it should be like key="[a,b]" or key=a,b It also
                        allows nested list/tuple values, e.g.
                        key="[(a,b),(c,d)]" Note that the quotation marks are
                        necessary and that no white space is allowed.
```

**Comment by RunningLeon** [comment]
@AllentDan 
`--model` for multiple paths of model files. `--shape` is optional since it could use `input_shape` from deploy_cfg.

**Comment by AllentDan** [comment]
> @AllentDan `--model` for multiple paths of model files. `--shape` is optional since it could use `input_shape` from deploy_cfg.

There is the only positional arg with nargs='+'. Meanwhile, it is the last positional arg and users can set multiple values to it.

File: docs/en/useful_tools.md [hunk 1]
```diff
@@ -144,8 +144,10 @@ python tools/profile.py \
     --device ${DEVICE} \
     --shape ${SHAPE} \
     --num-iter {NUM_ITER} \
-    --warmup {WARMUP}
-    --cfg-options ${CFG_OPTIONS}
+    --warmup {WARMUP} \
```
Comment (grimoire) [inline docs/en/useful_tools.md:147]: add a `$`

File: docs/en/useful_tools.md [hunk 2]
```diff
@@ -144,8 +144,10 @@ python tools/profile.py \
     --device ${DEVICE} \
     --shape ${SHAPE} \
     --num-iter {NUM_ITER} \
```
Comment (hanrui1sensetime) [inline docs/en/useful_tools.md:146]: May also need to add `$`

File: tools/profile.py [hunk 3]
```diff
@@ -120,11 +126,11 @@ def main():
     if not is_device_cpu:
         torch.backends.cudnn.benchmark = True
 
-    image_files = get_images(args.image_dir)
+    image_files = get_images(args.image_dir, args.img_ext)
     nrof_image = len(image_files)
     assert nrof_image > 0, f'No image files found in {args.image_dir}'
     logger.info(f'Found totally {nrof_image} image files in {args.image_dir}')
-    total_iters = args.num_iter + args.warmup
+    total_iters = (args.num_iter + args.warmup) * args.batch_size
```
Comment (grimoire) [inline tools/profile.py:133]: The variable name should be total_images instead of total_iters?

**Comment by lvhan028** [comment]
Please merge master to resolve readthedocs build error

**Review by hanrui1sensetime — APPROVED** [review]
lgtm.

**Review by AllentDan — COMMENTED** [review]
May resolve conflicts

**Review by grimoire — APPROVED** [review]

**Pull request merged**